src/term/mod.rs

```diff
@@ -1350,6 +1363,21 @@ impl ansi::Handler for Term {
     /// Set the window title
     #[inline]
     fn set_title(&mut self, title: &str) {
+        #[cfg(windows)]
+        {
+            // Windows - if using the winpty backend then Alacritty will
+            // recieve instructions to set the title to either blank, or
+            // e.g. the running process "- python".
+            //
+            // With the conpty backend the title is set correctly, so
+            // these same titles would be set from the conpty backend
+            // as e.g. "Alacritty" or "Alacritty - python"
+            //
+            // For this reason we disable setting the title when using
+            // the winpty backend.
```

Comment:
I don't see how completely disabling dynamic title support would be better than allowing WinPTY to set blank titles or something like `- python`. If people wish to have Alacritty as title, just disabling dynamic title support from the config would still be an option. Imho that is a better solution than adding a bunch of ugly workarounds just to disable a functionality which doesn't necessarily harm the user (and definitely doesn't cause any crashes).

Besides that, it sounds like the difference between WinPTY and ConPTY is just that ConPTY adds an `Alacritty` at the beginning. If that's all there is to it, why don't we just do the same with the WinPTY? That wouldn't require patching anything upstream, it seems like a pretty simple fix that should 'just work'?

Reply:
Fair point. I agree it's quite an ugly workaround.

> Besides that, it sounds like the difference between WinPTY and ConPTY is just that ConPTY adds an `Alacritty` at the beginning. If that's all there is to it, why don't we just do the same with the WinPTY?

I wondered about that - the titles are coming in from the term escape codes. Instead of a `dynamic_title_supported` field in `Term` we could have a `title_prefix` field. Which defaults empty, but in the `winpty` case we can set it to `"Alacritty"`. I'll do that.

Reply:
I don't like the idea of having a WinPTY specific hack like that in the config. If it makes sense to prepend `Alacritty` to all titles set by WinPTY, we should do exactly that, instead of shoving the responsibility on the user. Either it makes sense to do this (and we should do it), or it doesn't.

If users want to modify the title, then they can always set it explicitly, but we shouldn't be exposing workarounds to the user like that, when they've got no clue what the hell's going on.

Reply:
>  If it makes sense to prepend `Alacritty` to all titles set by WinPTY, we should do exactly that, instead of shoving the responsibility on the user.

Fully agree that the solution should be transparent to the user. I meant that this behaviour of appending `Alacritty` to all titles would need to be implemented in the `Term::set_title` code and based at runtime on either a `bool` or `String` configuring whether to do it at runtime.

As that also is quite ugly - for the moment I've just removed the Winpty workaround entirely.